src/client/Microsoft.Identity.Client/AppConfig/ConfidentialClientApplicationBuilder.cs

```diff
@@ -343,9 +345,25 @@ public ConfidentialClientApplicationBuilder WithTelemetryClient(params ITelemetr
                 Config.TelemetryClients = telemetryClients;
             }
 
+            TelemetryClientLogMsalVersion();
+
             return this;
         }
 
+        private void TelemetryClientLogMsalVersion()
+        {
+            MsalTelemetryEventDetails telemetryEventDetails = new MsalTelemetryEventDetails(TelemetryConstants.ConfigurationUpdateEventName);
+            telemetryEventDetails.SetProperty(TelemetryConstants.MsalVersion, MsalIdHelper.GetMsalVersion());
+
+            foreach (var telemetryClient in Config.TelemetryClients)

```

Comment:
Shouldn't the code in the foreach be factorized somewhere?
This seems like a lot of responsibility for the application.

cc: @jennyf19 as we were discussing the R9 telemetry.

Reply:
Thanks for the feedback @jmprieur, this is consistent with how we log in SAL. As discussed with @bgavrilMS, we can probably resolve this by adding a delegate to perform this task. @TimHannMSFT, thoughts on adding a delegate to iterate over ITelemetryClient list?

Reply:
I don't understand the delegate comment. My read of Jean-Marc's comment was to not have the concept of iterating over clients here but refactored into a method that does this in a standard way for all times you're emitting an event. I agree. See https://identitydivision.visualstudio.com/DevEx/_git/Microsoft-IdentityModel-S2S?path=/src/Microsoft.IdentityModel.S2S.Tokens/Telemetry/TelemetryExtension.cs for what we do in SAL.

Reply:
Thanks @TimHannMSFT, I have updated the code.